src/libraries/System.Net.NameResolution/tests/FunctionalTests/GetHostAddressesTest.cs

```diff
@@ -172,21 +172,31 @@ public async Task DnsGetHostAddresses_PreCancelledToken_Throws()
         }
 
         [OuterLoop]
-        [Fact]
+        [ConditionalFact]
         [ActiveIssue("https://github.com/dotnet/runtime/issues/43816")] // Race condition outlined below.
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/33378", TestPlatforms.AnyUnix)] // Cancellation of an outstanding getaddrinfo is not supported on *nix.
         public async Task DnsGetHostAddresses_PostCancelledToken_Throws()
         {
-            using var cts = new CancellationTokenSource();
+            const int NumberOfRequests = 100;
 
-            Task task = Dns.GetHostAddressesAsync(TestSettings.UncachedHost, cts.Token);
+            using CancellationTokenSource cts = new();
+            List<Task> tasks = new(capacity: NumberOfRequests);
+
+            for (int i = 0; i < NumberOfRequests; ++i)
+            {
+                Task task = Dns.GetHostAddressesAsync(TestSettings.UncachedHost, cts.Token);
+                tasks.Add(task);
+            }
 
-            // This test might flake if the cancellation token takes too long to trigger:
-            // It's a race between the DNS server getting back to us and the cancellation processing.
             cts.Cancel();
 
-            OperationCanceledException oce = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => task);
-            Assert.Equal(cts.Token, oce.CancellationToken);
+            try
+            {
+                await Task.WhenAll(tasks);
+                throw new SkipTestException("GetHostAddressesAsync should fail but it did not.");
```

Comment:
No one's going to notice this, especially in CI.  Could we just keep retrying until it does fail?  Folks will notice hangs ;-)

Reply:
Like a148365d8ad1dff4a06e1a28bf29b8a426ad003c?